### PR TITLE
Check if coursier upgrade is responsible of 1.4.0 failure

### PR DESCRIPTION
**Pull request by jvican**

### CODE REVIEW

File: launcher/src/main/scala/bloop/launcher/BloopServerVersion.scala [hunk 1]
```diff
@@ -12,13 +12,13 @@ case class BloopServerVersion(
 object BloopServerVersion {
   def apply(serverVersion: String, out: PrintStream): Option[BloopServerVersion] = {
     Version(serverVersion).items.toList match {
-      case (major: Version.Number) :: (minor: Version.Number) :: rest =>
+      case Version.Number(major) :: Version.Number(minor) :: rest =>
```
Comment (jvican) [inline launcher/src/main/scala/bloop/launcher/BloopServerVersion.scala:15]: @alexarchambault This is a revert of the changes I did for bloop to upgrade to the latest coursier. Was this a good upgrade to the new version API?

**Comment by jvican** [comment]
@tgodzik and I have confirmed that this causes the issue some people have experienced in 1.4.0 https://github.com/scalacenter/bloop/issues/1276 I will cut a release tonight

File: launcher/src/main/scala/bloop/launcher/BloopServerVersion.scala (hunk 1 above)
Comment (alexarchambault) [reply launcher/src/main/scala/bloop/launcher/BloopServerVersion.scala:15]: What problem are you running into with these changes?

Comment (jvican) [reply launcher/src/main/scala/bloop/launcher/BloopServerVersion.scala:15]: We don't know yet, we have bisected the commit history and gotten to these changes. I think the upgrade is somehow affecting the automatic SemanticDB resolution.

**Comment by ivanopagano** [comment]
Can you confirm if this fix will allow the user to get back bash-completions?

Was that eventually included in the 1.4.0 cut?

**Comment by jvican** [comment]
Bash completions are part of the installer but you need to uninstall and install back again to get them to work because there was a bug in the home brew recipe.

**Comment by olafurpg** [comment]
I was experimenting with bloop-launcher 1.4.0 and hit on the following error
```
[error] Exception in thread "bsp-client-to-server" Exception in thread "main" java.lang.NoSuchMethodError: java.nio.ByteBuffer.flip()Ljava/nio/ByteBuffer;
[error] 	at bloop.launcher.bsp.BspBridge.copyContents(BspBridge.scala:290)
[error] 	at bloop.launcher.bsp.BspBridge.forwardStreamContents(BspBridge.scala:275)
[error] 	at bloop.launcher.bsp.BspBridge.$anonfun$wireBspConnectionStreams$1(BspBridge.scala:197)
[error] 	at bloop.bloopgun.core.Shell$$anon$2.run(Shell.scala:155)
```
Downgrading to v1.4.0-RC1-235-3231567a fixes the problem. I wonder if this could be the root cause of the 1.4.0 errors?

**Comment by tgodzik** [comment]
> I was experimenting with bloop-launcher 1.4.0 and hit on the following error
> 
> ```
> [error] Exception in thread "bsp-client-to-server" Exception in thread "main" java.lang.NoSuchMethodError: java.nio.ByteBuffer.flip()Ljava/nio/ByteBuffer;
> [error] 	at bloop.launcher.bsp.BspBridge.copyContents(BspBridge.scala:290)
> [error] 	at bloop.launcher.bsp.BspBridge.forwardStreamContents(BspBridge.scala:275)
> [error] 	at bloop.launcher.bsp.BspBridge.$anonfun$wireBspConnectionStreams$1(BspBridge.scala:197)
> [error] 	at bloop.bloopgun.core.Shell$$anon$2.run(Shell.scala:155)
> ```
> 
> Downgrading to v1.4.0-RC1-235-3231567a fixes the problem. I wonder if this could be the root cause of the 1.4.0 errors?

That's weird, Ihaven't seen that issue and `flip` should be avlaiable both in JDK 8 and 11

**Comment by tgodzik** [comment]
So I did some digging and it seems that 1.4.0 works fine on JDK 11, but it doesn't on JDK 8. I can reproduce 100% with JDK 8.

**Comment by jvican** [comment]
This attempt at downgrading bloop is a red herring, real fix in https://github.com/scalacenter/bloop/pull/1278

**Pull request closed without merging**